server/test/test_json.go

```diff
@@ -0,0 +1,42 @@
+package test
+
+import "encoding/json"
+
+type spec struct {
+	Selector   Selector    `json:"selector"`
+	Name       string      `json:"name,omitempty"`
+	Assertions []Assertion `json:"assertions"`
+}
+
+type namedAssertion struct {
+	Name       string
+	Assertions []Assertion
+}
+
+type keyValueSpec struct {
```

Comment:
I think it would be better to version the entire `TestSpec` json instead of just the things that change, like here: https://github.com/kubeshop/tracetest/blob/main/server/model/trigger_json.go

This makes it easier to keep updating the json, adding new versions, with a standardized approach